Create policy and policy request

Smightym8 opened the pull request:
- **feat(policy): implement creating policy**
  - Added CreatePolicyEndpoint.
  - Implemented policy creation in PolicyCreateService.
  - Implemented id generation and persisting policy in PolicyRepository.
  - Implemented unit and integration tests.
  

- **refactor(domain): change relationship between policy and smart meter**
  - Changed relationship between policy and smart meter from many to many to one to many.
  - Updated domain model in README.md to reflect the change.
  - Removed Location and HouseHoldSize from Policy.
  - Updated related code.
  

- **feat(policy): implement validating Policy**
  - Implemented validating if location resolution matches with the location specified in the metadata.
  - Added unit tests.
  

- **feat(policyrequest): implement create policy request**
  - Implemented CreatePolicyRequestEndpoint.
  - Implemented creating policy request in PolicyRequestCreateService.
  - Implemented persisting policy request in PolicyRequestRepository.
  - Implemented unit and integration tests.
  - Added InsufficientLocationDataException to GlobalExceptionHandler.
  - Formatted code with dotnet format.
  

- **refactor(db): change column names in entity configurations**
  - Changed entity names that are hardcoded in the entity configurations to be camel case.

Smightym8 committed:
feat(policy): implement creating policy

- Added CreatePolicyEndpoint.
- Implemented policy creation in PolicyCreateService.
- Implemented id generation and persisting policy in PolicyRepository.
- Implemented unit and integration tests.

Smightym8 committed:
refactor(domain): change relationship between policy and smart meter

- Changed relationship between policy and smart meter from many to many to one to many.
- Updated domain model in README.md to reflect the change.
- Removed Location and HouseHoldSize from Policy.
- Updated related code.

Smightym8 committed:
feat(policy): implement validating Policy

- Implemented validating if location resolution matches with the location specified in the metadata.
- Added unit tests.

Smightym8 committed:
feat(policyrequest): implement create policy request

- Implemented CreatePolicyRequestEndpoint.
- Implemented creating policy request in PolicyRequestCreateService.
- Implemented persisting policy request in PolicyRequestRepository.
- Implemented unit and integration tests.
- Added InsufficientLocationDataException to GlobalExceptionHandler.
- Formatted code with dotnet format.

Smightym8 committed:
refactor(db): change column names in entity configurations

- Changed entity names that are hardcoded in the entity configurations to be camel case.

Smightym8 committed:
test: update assertion to use Has.Count.EqualTo for collection count check

Essiga commented on src/SMAIAXBackend.Application/DTOs/PolicyCreateDto.cs:
```diff
@@ -0,0 +1,21 @@
+    [Required]
+    public LocationResolution LocationResolution { get; set; } = locationResolution;
+    [Required]
+    public decimal Price { get; set; } = price;
```
Maybe out of scope but not sure how to properly handle price localization. Maybe using PriceInEuro would be better here

Smightym8 replied:
The [Money Pattern](https://martinfowler.com/eaaCatalog/money.html) would be a perfect fit for this. I created a task on the project board.

Essiga commented on src/SMAIAXBackend.Application/Services/Implementations/SmartMeterListService.cs:
```diff
@@ -43,14 +44,15 @@ public async Task<SmartMeterOverviewDto> GetSmartMeterByIdAndUserIdAsync(Guid sm
     }
 
-    private static SmartMeterOverviewDto SmartMeterOverviewDtoFromSmartMeter(SmartMeter smartMeter)
+    private async Task<SmartMeterOverviewDto> SmartMeterOverviewDtoFromSmartMeter(SmartMeter smartMeter, UserId userId)
```
why not make this a method on SmartMeterOverviewDto? That way it could be used more universally

Essiga reviewed (approved):
🦈

Smightym8 committed:
refactor(smartmeter): move SmartMeterOverviewDtoFromSmartMeter method to SmartMeterOverviewDto as static method

- Moved `SmartMeterOverviewDtoFromSmartMeter` from `SmartMeterListService` to `SmartMeterOverviewDto`
- Renamed method to `FromSmartMeter`
- Improved cohesion by keeping conversion logic within `SmartMeterOverviewDto`
- Changed method to be static for direct class-level access
